apps/keyboard/js/settings/panel_base.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+/* global ViewBase */
+
+/*
+ * The base class of a Panel/Dialog, as a subclass of ViewBase.
+ *
+ * Panels/Dialogs may have child Views which we propogate beforeShow/show/
+ * beforeHide/hide event hooks and make sure the asynchronousness is preserved.
+ *
+ * Additionally, we accept |options| parameter at beforeShow for initialization
+ * of a Panel/Dialog
+ */
+ 
+
+(function(exports) {
+
+var PanelBase = function() {
+};
+
+PanelBase.prototype = Object.create(ViewBase.prototype);
+
+PanelBase.prototype.CONTAINER_ID = '';
+
+PanelBase.prototype.container = null;
+
+// A map from view names to Views.
+PanelBase.prototype.views = {};
```

Comment:
Or, `= null;` here and have the constructor to create this object when it is called.